lua/nvim-treesitter-textobjects/shared.lua

```diff
@@ -220,47 +214,80 @@ function M.find_best_range(bufnr, capture_string, query_group, filter_predicate,
   return best
 end
 
+---@param range Range4
+---@param row integer
+---@param col integer
+---@return boolean
+local function is_in_range(range, row, col)
+  local start_row, start_col, end_row, end_col = unpack(range) ---@type integer, integer, integer, integer
+  end_col = end_col - 1
+
+  local is_in_rows = start_row <= row and end_row >= row
+  local is_after_start_col_if_needed = true
+  if start_row == row then
+    is_after_start_col_if_needed = col >= start_col
+  end
+  local is_before_end_col_if_needed = true
+  if end_row == row then
+    is_before_end_col_if_needed = col <= end_col
+  end
+  return is_in_rows and is_after_start_col_if_needed and is_before_end_col_if_needed
+end
+
+---@param range1 Range4
+---@param range2 Range4
+---@return boolean
+local function contains(range1, range2)
+  return is_in_range(range1, range2[1], range2[2]) and is_in_range(range1, range2[3], range2[4])
+end
+
+---@param range Range6
+---@return Range4
+function M.range6_range4(range)
```

Comment:
Not 100% happy with the name here (I know it's a temporary utility function, but old habits...)

Better (explicit) would be `range6_to_range4`, though that's a bit long maybe. Just `torange4` or `make_range4`?